scripts/scons_plugin/systemdeps.py

```diff
@@ -0,0 +1,100 @@
+import os
+
+def GeneratePkgConfig(env, build_dir, filename, prefix, libdir, name, desc, url, version):
+    target = os.path.join(build_dir, filename)
+
+    dep_list = env.get('PKG_CONFIG_DEPS', [])
+
+    lib_list = \
+      ['roc'] + \
+      env.get('PKG_CONFIG_LIBS', [])
+
+    incdir_list = \
+      [prefix+'/include'] + \
+      env.get('PKG_CONFIG_CPPPATH', [])
+
+    libdir_list = \
+      [libdir] + \
+      env.get('PKG_CONFIG_LIBPATH', [])
+
+    src = """prefix=%(prefix)s
+exec_prefix=${prefix}
+
+Name: %(name)s
+Requires: %(deps)s
+Version: %(version)s
+Description: %(desc)s
+URL: %(url)s
+
+Libs: %(libdirs)s %(libs)s
+Cflags: %(incdirs)s
+""" % { 'prefix': prefix,
```

Comment:
May I ask you to use `format()` instead of `%` operator in all new functions please? I almost broke my head trying to understand what does `s` suffix mean.

> format(*args, **kwargs)
> This method of string formatting is the new standard in Python 3, and should be preferred to the % formatting described in String Formatting Operations in new code.

https://docs.python.org/2/library/stdtypes.html#str.format

```
Python 2.7.18.6 (default, Jan  1 1980, 00:00:00) 
[GCC 11.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> """ xxx={xxx} """.format(xxx=123)
' xxx=123 '

Python 3.10.9 (main, Dec  6 2022, 18:44:57) [GCC 11.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> """ xxx={xxx} """.format(xxx=123)
' xxx=123 '
```

Reply:
Done.